software/contrib/egressus_melodiam.py

```diff
@@ -0,0 +1,969 @@
+often caused by under-samping a wave form.
+
+"""
+# Minimum allowed time between incoming 16th note clocks. Smaller values are too fast for poor ole micropython to keep up
+MIN_CLOCK_TIME_MS = 50
+
+# Maximum allowed time between incoming 16th note clocks. This is 2 BPM, larger value causes memory issues unless the next two vals are halfed
+MAX_CLOCK_TIME_MS = 3750
+
+# Max sample rate and clock divisor to keep required buffer sizes within memory limits
+MAX_SAMPLE_RATE = 32
+MAX_OUTPUT_DENOMINATOR = 8
+
+# Calculate maximum sample buffer size required
+SLEW_BUFFER_SIZE_IN_SAMPLES = int(
+    (MAX_CLOCK_TIME_MS / 1000) * MAX_SAMPLE_RATE * MAX_OUTPUT_DENOMINATOR
+)
+
+# Keep these in for future debugging during feature expansion
+DEBUG_VERBOSE = 3
+DEBUG_LOTS = 2
+NO_DEBUG = 0
+DEBUG_MODE = NO_DEBUG
+
+# Attempt to avoid knob hysteresis
+KNOB_CHANGE_TOLERANCE = 0.999
+
+# Set the maximum CV voltage using a global config value
+# Important: Needs firmware v0.12.1 or higher
+MAX_CV_VOLTAGE = europi_config["max_output_voltage"]
+
+MAX_STEP_LENGTH = 32
+
+# Diff between incoming clocks are stored in the FiFo buffer and averaged
+# Averaging over 5 values seems to deal with wonky clocks quite well
+CLOCK_DIFF_BUFFER_LEN = 5
+
+# If the clock rate changes more than this, trigger a recalculation.
+# Avoids wonky waves when wonky clocks are used.
+MIN_CLOCK_CHANGE_DETECTION_MS = 100
+
+# Slightly quicker way to get integers from boolean values
+BOOL_DICT = { False: 0, True: 1}
+
+# Wave shape bit arrays
+WAVE_SHAPE_IMGS = [
+    bytearray(b'\xfe\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x82\x10\x83\xf0'), # stepUpStepDown
+    bytearray(b'\x00\x00\x06\x00\x05\x00\t\x00\t\x00\x10\x80\x10\x80 @ @@ @ \x80\x10'), # linspace (tri)
+    bytearray(b'0\x00(\x10D\x10D\x10D D\x10\x82 \x82 \x82 \x82 \x81@\x01\xc0'), # smooth (sine)
+    bytearray(b'\x04\x00\x04\x00\x06\x00\x06\x00\n\x00\t\x00\t\x00\x10\x80 \x80 @@ \x80\x10'), # expUpexpDown
+    bytearray(b'\x0c\x00\x12\x00\x12\x00"\x00"\x00A\x00A\x00@\x80@\x80\x80@\x80 \x80\x10'), # sharkTooth
+    bytearray(b'\x04\x00\x05\x00\x04\x80\x08\x80\x08@\x08@\x10 \x10 \x10   @\x10\x80\x00'), # sharkToothReverse
+    bytearray(b'\x03\xf0\x0c\x100\x10 \x10@\x10@\x10@\x10\x80\x10\x80\x10\x80\x10\x80\x10\x80\x10'), # logUpStepDown
+    bytearray(b'\xff\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80\x80@\x80@\x80 \x80\x10'), # stepUpExpDown
+]
+
+class EgressusMelodiam(EuroPiScript):
+    def __init__(self):
+
+        # Overclock the Pico for improved performance.
+        machine.freq(250_000_000)
+
+        # Initialize variables
+        self.clockStep = 0
+        self.stepPerOutput = [0, 0, 0, 0, 0, 0]
+        self.nextStepPerOutput = [0, 0, 0, 0, 0, 0]
+        self.CvPattern = 0
+        self.resetTimeout = MAX_CLOCK_TIME_MS
+        self.screenRefreshNeeded = True
+        self.shreadedVis = False
+        self.shreadedVisClockStep = 0

```

Comment:
[question]
What does `shreaded` mean in this context? Is it an acronym? Or just a term I've never seen before?

Reply:
renamed